crates/ruff_dev/src/generate_options.rs

```diff
@@ -1,9 +1,68 @@
 //! Generate a Markdown-compatible listing of configuration options for `pyproject.toml`.
 //!
 //! Used for <https://docs.astral.sh/ruff/settings/>.
-use itertools::Itertools;
+use std::fmt::Write;
+
 use ruff_workspace::options::Options;
-use ruff_workspace::options_base::{OptionEntry, OptionField};
+use ruff_workspace::options_base::{OptionField, OptionSet, OptionsMetadata, Visit};
+
+pub(crate) fn generate() -> String {
+    let mut output = String::new();
+    generate_set(&mut output, &Set::Toplevel(Options::metadata()));
+
+    output
+}
+
+fn generate_set(output: &mut String, set: &Set) {
+    writeln!(output, "### {title}\n", title = set.title()).unwrap();
```

Comment:
`writeln!` with trailing `\n` looks strange

Reply:
Agree but everything else feels verbose just to get two newlines

Reply:
(I typically do two `writeln` for this but don't feel strongly.)